src/dataframe/dataframe.jl

```diff
@@ -749,6 +759,20 @@ merge!(df, df2)  # column z is added, column id is overwritten
 """
 function Base.merge!(df::DataFrame, others::AbstractDataFrame...)
     for other in others
+        d = diff_indices(index(other), index(df))
+        #=
```

Comment:
This should go to the docstring for `append!`. Also, better find another name since it doesn't follow the signature of the generic `append!`.